src/files/files.service.ts

```diff
@@ -1,21 +1,21 @@
-import {HttpException, HttpStatus, Injectable} from '@nestjs/common';
+import { Injectable, InternalServerErrorException } from '@nestjs/common';
 import * as path from 'path';
 import * as fs from 'fs';
 import * as uuid from 'uuid';
 
 @Injectable()
 export class FilesService {
-    async createFile(file): Promise<string> {
-        try {
-            const fileName = uuid.v4() + '.jpg';
-            const filePath = path.resolve(__dirname, '..', 'static')
-            if (!fs.existsSync(filePath)) {
-                fs.mkdirSync(filePath, {recursive: true})
-            }
-            fs.writeFileSync(path.join(filePath, fileName), file.buffer)
-            return fileName;
-        } catch (e) {
-            throw new HttpException('Error when write file', HttpStatus.INTERNAL_SERVER_ERROR)
-        }
+  async createFile(file): Promise<string> {
+    try {
+      const fileName = uuid.v4() + '.jpg';
+      const filePath = path.resolve(__dirname, '..', 'static');
+      if (!fs.existsSync(filePath)) {
+        fs.mkdirSync(filePath, { recursive: true });
+      }
+      fs.writeFileSync(path.join(filePath, fileName), file.buffer);
```

Comment:
Let's avoid any **sync** I/O operations, because I/O is veeeery slow by nature — and it blocks our main thread.
Instead, we have to use **async** APIs. Node has two options for async fs access:
1. [Callback API](https://nodejs.org/docs/latest-v16.x/api/fs.html#callback-api) — it's ok, but it's old.
2. [Promises API](https://nodejs.org/docs/latest-v16.x/api/fs.html#promises-api) — let's use this one as newer.